cpp/src/rolling/rolling.cu

```diff
@@ -76,9 +78,10 @@ process_rolling_window(column_device_view input,
     if (!has_nulls) {
       count = end_index - start_index;
     } else {
-      for (size_type j = start_index; j < end_index; j++) {
-        if (input.is_valid(j)) { count++; }
-      }
+      count = thrust::count_if(thrust::seq,
+                               thrust::make_counting_iterator(start_index),
+                               thrust::make_counting_iterator(end_index),
+                               [&input](auto i) { return input.is_valid(i); });
```

Comment:
To avoid the `is_nullable()` check on every access to is_valid...

```suggestion
                               [&input](auto i) { return input.is_valid_no_check(i); });
```

Reply:
Thanks, I had to look that up. `has_nulls` is actually a one-time reading of `input.has_nulls()`, so it should be safe to call `is_valid_no_check()`.
Super. I'll change this and test it.